vital/range/iterator_range.h

```diff
@@ -0,0 +1,43 @@
+// This file is part of KWIVER, and is distributed under the
+// OSI-approved BSD 3-Clause License. See top-level LICENSE file or
+// https://github.com/Kitware/kwiver/blob/master/LICENSE for details.
+
+#ifndef VITAL_ITERATOR_RANGE_H
+#define VITAL_ITERATOR_RANGE_H
+
+/// \file Class to wrap two iterators in a range-friendly way.
+
+namespace kwiver {
+
+namespace vital {
+
+namespace range {
+
+// ----------------------------------------------------------------------------
+/// Wraps two iterators into a range.
+template < class Iterator >
+class iterator_range
+{
+public:
+  iterator_range( Iterator const& begin, Iterator const& end )
+    : m_begin{ begin }, m_end{ end } {}
+  iterator_range( Iterator&& begin, Iterator&& end )
+    : m_begin{ begin }, m_end{ end } {}
+
+  Iterator
+  begin() const { return m_begin; }
+  Iterator
+  end() const { return m_end; }
```

Comment:
```suggestion
  Iterator begin() const { return m_begin; }
  Iterator end() const { return m_end; }
```

Reply:
Is there a reason for the different style inside and outside a class?